src/response.zig

```diff
@@ -161,20 +166,23 @@ pub const Response = struct {
         buf[1] = '\n';
         const len = 2 + std.fmt.formatIntBuf(buf[2..], data.len, 16, .upper, .{});
         buf[len] = '\r';
-        buf[len+1] = '\n';
-        try stream.writeAll(buf[0..len+2]);
+        buf[len + 1] = '\n';
+        try stream.writeAll(buf[0 .. len + 2]);
         try stream.writeAll(data);
     }
 
     pub fn writer(self: *Response) Writer.IOWriter {
-        return .{ .context = Writer.init(self) };
+        if (self.use_writer == null) {
```

Comment:
this is the only interesting part of this PR

the rest of it seems to be zig fmt doing its thing